### DISCUSSION TITLE
Sync between devices

### DISCUSSION THREAD

**Discussion by hmbenhaim** [isar/isar#708] [Ideas]
Hi everyone,

I saw an open issue about synchronization database across devices

This something that I wish isar had
Realm as such thing but i think Isar is a better fit for some usecases and isar support web

I'm just wondering if this is something that is going to be soon and if so how it's going to be implemented (by dB in the cloud like realm or web sockets or any other solution)

Thank you

**Comment by CntrlX** [comment]
Hi can you tell me an exact use case, i would like to develop this feature and i am thinking on how to structure it. Your answer would help me do thjs greatly

**Reply by shiyewuxian** [reply]
Hi @CntrlX ,

I would like to share my user case for your reference.
There are 2 scenarios:
1. like realm or couchbase or objectBox, devices like mobiles or embedded devices could build direct connection (e.g. peer to peer). Then based on the connection, whenever there're changes, devices will send the changes part (better to be changes prats no the whole data, otherwise the network loads will be huge). Devices receive the changes and put them into a transaction or the queue, then process and update local db. For example, a chat app like whatsapp, user do not need to send the messages to backend server.
2. Sync between client and backend server. From my understanding, there should be a server running as a backend and client build connection with the server by websocket/Http3/others if suitable. Then shared changes in 2 ways.

**Reply by ogzs** [reply]
I would suggest using couchdb as the backend, because it has a great "_changes" feed, and was designed for replication. It's open-source, we just have to implement their replication protocol.
I would also like to develop this feature

**Reply by shiyewuxian** [reply]
@ogzs 
Hi ogzs, thanks for your response. I checked some more solutions, for example, [isar_crdt](https://github.com/davidlondono/isar_crdt), which is however not maintained any longer. 

Really looking forward to your great solution.
